src/colorizer/Dataset.ts

```diff
@@ -76,17 +77,24 @@ export default class Dataset {
   /** Lookup from a global index of an object to the raw segmentation ID in the
    * frame/image where it appears. */
   public segIds?: Uint32Array | null;
+
   /**
-   * Lookup table from a frame number to the offsets with which to read data
-   * from globally-indexed arrays for objects on that frame. Note that this
-   * assumes that all objects in a frame have segmentation IDs that are
-   * monotonically increasing (e.g. skips no values), and that a frame's objects
-   * all have adjacent indices in the global array.
+   * Maps from a frame number to a lookup table used to get the global ID of a
+   * segmentation ID in that frame.
+   *
+   * For segmentation ID `segId` at time `t`, the global ID is given by:
+   *
+   * ```
+   * const globalIdLut = frameToGlobalIdLookup[t];
+   * const globalId = globalIdLut.lut[segId - globalIdLut.minSegId] - 1;
+   * ```
    *
-   * For any object with segmentation ID `segId` at time `t`, the global
-   * index of that object is given by `segId + frameToIdOffsets[t]`.
+   * The global ID is `NaN` or `-1` if there is no global ID that matches that
+   * segmentation ID, such as when rows are missing in the dataset.
+   *
+   * See `GlobalIdLookupInfo` for more details.
    */
-  public frameToIdOffset: Uint32Array | null;
+  public frameToGlobalIdLookup: Map<number, GlobalIdLookupInfo> | null;
```

Comment:
Why a `Map` and not an array? If we expect frame numbers to be dense and begin at 0, an array would store and retrieve them more efficiently.

Reply:
There *is* an issue ticket in the backlog for datasets that have frame numbers that don't start at 0, I believe. #203 
I think switching this to an array is fine with me, but it would require a refactor on the vole-core side. I'm going to merge as-is but I'll open an issue ticket about refactoring this later.

Reply:
Created as #655